model/host/hostutil.go

```diff
@@ -744,70 +744,60 @@ const jasperDialTimeout = 15 * time.Second
 // JasperClient returns a remote client that communicates with this host's
 // Jasper service.
 func (h *Host) JasperClient(ctx context.Context, env evergreen.Environment) (jasper.RemoteClient, error) {
-	if h.LegacyBootstrap() || h.LegacyCommunication() {
+	if (h.LegacyBootstrap() || h.LegacyCommunication()) && h.NeedsReprovision != ReprovisionToLegacy {
```

Comment:
should this become a method on host documents?

Reply:
See other comment.